pulp_container/app/serializers.py

```diff
@@ -358,23 +358,26 @@ def validate(self, data):
         method checks if the tag exists within the repository.
         """
         new_data = super().validate(data)
+        new_data["manifest"] = get_manifest(new_data)
+        return new_data
 
-        try:
-            manifest = models.Manifest.objects.get(
-                pk__in=new_data["latest_version"].content.all(), digest=new_data["digest"]
-            )
-        except models.Manifest.DoesNotExist:
-            raise serializers.ValidationError(
-                _(
-                    "A manifest with the digest '{}' does not "
-                    "exist in the latest repository version '{}'".format(
-                        new_data["digest"], new_data["latest_version"]
-                    )
+
+def get_manifest(data):
```

Comment:
extracting the repeated code into a separate function and leaving it like this in between all other serializers' code makes it hard to read because it gets 'lost', you need to logically place it.
you can define a mixin that would validate repo_version and manifest or you can just let it be and leave this try/excpet repeated twice. Sometimes it is more readable to have the code repeated rather then adding mixins and have multi-inheritance for the sake of saving several lines of code.

Reply:
I am wondering if this Manifest lookup can be rephrased with a HiddenField, as well as the latest_version [0]. But that would be refactoring and we should not do that along with a feature PR. I agree with @ipanova that sometimes keeping the things where they happen to not have to scoll up and down to understand makes reading easier.
At the very least i am quite unhappy with the function name, because in django all the "get_<an object>" functions raise `DoesNotExist` and this one will raise `ValidationError`.

[0] https://www.django-rest-framework.org/api-guide/fields/#hiddenfield